src/pepr/policies/storage.ts

```diff
@@ -24,39 +25,73 @@ When(a.Pod)
   .IsCreatedOrUpdated()
   .Mutate(markExemption(Policy.RestrictVolumeTypes))
   .Validate(request => {
-    // List of allowed volume types
-    const allowedVolumeTypes = [
-      "configMap",
-      "csi",
-      "downwardAPI",
-      "emptyDir",
-      "ephemeral",
-      "persistentVolumeClaim",
-      "projected",
-      "secret",
-    ];
-
     if (isExempt(request, Policy.RestrictVolumeTypes)) {
       return request.Approve();
     }
 
-    // Check all volumes in the pod spec, if any
-    for (const volume of volumes(request)) {
-      // Get the volume type, which will be the only key in the volume object other than "name"
-      const volumeType = Object.keys(volume).find(key => key !== "name") || "unknown";
+    const [isValid, invalidVolume] = validateVolumeTypes(volumes(request));
 
-      // If the volume type is not in the allowed list, deny the request
-      if (!allowedVolumeTypes.includes(volumeType)) {
-        return request.Deny(
-          `Volume ${volume.name} has a disallowed volume type of '${volumeType}'.`,
-        );
-      }
+    if (!isValid && invalidVolume) {
+      return request.Deny(
+        `Volume '${invalidVolume.name}' has a disallowed volume type of '${invalidVolume.type}'.`,
```

Comment:
Single quotes added here are causing the e2e test failure - will just need to update the expected text in the e2e test (or revert the single quotes).